core/state/snapshot/conversion.go

```diff
@@ -355,3 +355,22 @@ func stdGenerate(db ethdb.Database, in chan trieKV, out chan common.Hash) {
 	}
 	out <- root
 }
+
+func stackTrieGenerate(db ethdb.Database, in chan trieKV, out chan common.Hash) {
+	commit := db != nil
+	if db == nil {
+		db = rawdb.NewMemoryDatabase()
+	}
+	triedb := trie.NewDatabase(db)
+	t := trie.NewStackTrie()
+	for leaf := range in {
+		t.TryUpdate(leaf.key[:], leaf.value)
+	}
+	var root common.Hash
+	if !commit {
+		root = t.Hash()
+	} else {
+		root = t.Commit(triedb)
```

Comment:
Please also commit the `triedb` here. Otherwise, we just discard all committed entries silently.